contracts/access/sources/ownership_transfer/delayed.move

```diff
@@ -0,0 +1,216 @@
+/// Time-locked wrapper that enforces a configurable delay between scheduling and executing
+/// the transfer of a capability.
+///
+/// Owners wrap the capability and must queue a transfer via the scheduling helpers.
+/// After scheduling, the owner must wait for the deadline to pass before executing,
+/// guaranteeing an on-chain lead time for sensitive capability moves.
+///
+/// Unwrapping follows the same delayed pattern, ensuring the capability cannot be reclaimed
+/// without respecting the delay period.
+module openzeppelin_access::delayed_transfer;
+
+use sui::clock;
+use sui::dynamic_object_field as dof;
+use sui::event;
+
+const WRAPPED_FIELD: vector<u8> = b"wrapped";
+
+#[error(code = 0)]
+const ETransferAlreadyScheduled: vector<u8> = b"Transfer already scheduled.";
+#[error(code = 1)]
+const ENoPendingTransfer: vector<u8> = b"No pending transfer.";
+#[error(code = 2)]
+const EDelayNotElapsed: vector<u8> = b"Delay has not elapsed.";
+#[error(code = 3)]
+const EWrongPendingAction: vector<u8> = b"Pending action mismatch.";
+
+/// Wrapper object that delays transfers by at least `min_delay_ms` after scheduling.
+public struct DelayedTransferWrapper<phantom T: key + store> has key {
+    id: UID,
+    min_delay_ms: u64,
+    pending: option::Option<PendingTransfer>,
+}
+
+/// Snapshot of a scheduled transfer or unwrap, including the execution time.
+/// A non-existing recipient means an unwrap is scheduled.
+public struct PendingTransfer has drop, store {
+    recipient: option::Option<address>,
+    execute_after_ms: u64,
+}
+
+/// Emitted when a delayed transfer is scheduled.
+public struct TransferScheduled has copy, drop {
+    wrapper_id: ID,
+    current_owner: address,
+    new_owner: address,
+    execute_after_ms: u64,
+}
+
+/// Emitted when an unwrap is scheduled.
+public struct UnwrapScheduled has copy, drop {
+    wrapper_id: ID,
+    current_owner: address,
+    execute_after_ms: u64,
+}
+
+/// Emitted when a delayed transfer is executed.
+public struct OwnershipTransferred has copy, drop {
+    wrapper_id: ID,
+    previous_owner: address,
+    new_owner: address,
+}
+
+// === Wrap / unwrap / borrow ===
+
+/// Wrap a capability/object in a delayed transfer wrapper with the desired minimum delay. The
+/// capability is tucked under a dynamic field so its object ID remains discoverable.
+public fun wrap<T: key + store>(
+    cap: T,
+    min_delay_ms: u64,
+    ctx: &mut TxContext,
+): DelayedTransferWrapper<T> {
+    let mut wrapper = DelayedTransferWrapper {
+        id: object::new(ctx),
+        min_delay_ms,
+        pending: option::none(),
+    };
+    dof::add(&mut wrapper.id, WRAPPED_FIELD, cap);
+    wrapper
+}
+
+/// Borrow the wrapped capability immutably—useful for inspection without touching the schedule.
+public fun borrow<T: key + store>(wrapper: &DelayedTransferWrapper<T>): &T {
```

Comment:
If you want this to be supported in PTB context, you need to allow borrowing _by value_. See [borrowing items from Kiosk](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/packages/sui-framework/sources/kiosk/kiosk.move#L600).

Reply:
Otherwise you'd force everyone using it to implement a custom function in their packages to support it. This will be changed eventually, but currently there's no other way to provide PTB support.